src/system/user.cr

```diff
@@ -7,23 +7,25 @@ class System::User
 
   extend Crystal::System::User
 
-  getter name : String
+  getter username : String
   getter user_id : LibC::UidT
   getter group_id : LibC::GidT
+  getter name : String
   getter directory : String
   getter shell : String
 
   def_equals_and_hash @user_id
 
-  private def initialize(@name, @user_id, @group_id, @directory, @shell)
+  private def initialize(@username, @user_id, @group_id, gecos, @directory, @shell)
+    @name = gecos.split(",").first
```

Comment:
This extraction from gecos should be in the platform-specific code. `System::User.new` should directly accept a `@name` argument.